lib/hystrix-stats.js

```diff
@@ -41,34 +35,45 @@ hystrixStream.resume();
  */
 class HystrixStats {
   constructor (circuit) {
-    const _readableStream = new Readable({
+    this._readableStream = new Readable({
       objectMode: true,
       read () {}
     });
 
     // Listen for the stats's snapshot event
     circuit.status.on('snapshot', function snapshotListener (stats) {
       // when we get a snapshot push it onto the stream
-      _readableStream.push(
-        Object.assign({},
-          {
-            name: circuit.name,
-            closed: circuit.closed,
-            group: circuit.group,
-            options: circuit.options
-          }, stats));
-    });
+      if (this._readableStream) {
```

Comment:
`_readableStream` is never unset in `shutdown()` so I think this will always be true.
Maybe save the listener on the instance and add`circuit.status.removeListener('snapshot', this._ snapshotListener)` to `shutdown()`. You don't need to keep the bind if you have the readable stream as a local const too.

Reply:
I think it makes more sense to add a shutdown() method to Status which removes all listeners, and call it from shutdown() in the circuit. As for _readableStream not ever being unset, this is true. However in my tests I sometimes get the error TypeError: Cannot read property 'push' of undefined which I find quite odd. The only thing I can figure is that a snapshot event is being emitted before the constructor completes. I'm going to poke around on this a bit.